### ISSUE TITLE
Formatting issue with currying function

### ISSUE THREAD

**Issue by manojeeva** [Dart-Code/Dart-Code#2133]
Issue Type: <b>Bug</b>
 
Closing Tag does have extra intent and that leads to an issue When collapse it does not show the closing tag. 

Before the Collapse and it is formatted.
<img width="831" alt="Screen Shot 2019-11-28 at 13 16 41" src="https://user-images.githubusercontent.com/46526800/69787006-926aad80-11e1-11ea-8125-38c181998975.png">

After the Collapse

<img width="790" alt="Screen Shot 2019-11-28 at 13 16 56" src="https://user-images.githubusercontent.com/46526800/69787040-a6161400-11e1-11ea-9007-98729341c78c.png">

Extension version: 3.6.0
VS Code version: Code 1.40.2 (f359dd69833dd8800b54d458f6d37ab7c78df520, 2019-11-25T14:52:45.129Z)
OS version: Darwin x64 19.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5287U CPU @ 2.90GHz (4 x 2900)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|8.00GB (0.31GB free)|
|Process Argv|-psn_0_57358|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by DanTup** [comment]
> When collapse it does not show the closing tag

Do you mean the closing `}` is "inside" the collapsed range instead of outside? I don't think is related to the indenting, I think this is always the case.

VS Code [doesn't support folding partial lines](https://code.visualstudio.com/api/references/vscode-api#FoldingRange) with columns, only lines. This means we have to decide whether to include the last line or not inside the range. If we include it, then it's not rendered when collapsed (as shown here), and if we don't include it then the collapsed methods would have an additional line for just the closing brace.

Eg. What I'd like to have is:

```dart
myFunction() { [...] }
```

But actually we only have:

```dart
myFunction() { [...]
```

or:

```dart
myFunction() { [...]
}
```

I'm not really sure if the bottom one is a great improvement over the current one, since it takes more space in the editor just for the closing brace.

**Comment by DanTup** [comment]
There are some related VS Code issues about this here:

https://github.com/microsoft/vscode/issues/3352#issuecomment-513696419
https://github.com/microsoft/vscode/issues/70794

**Comment by manojeeva** [comment]
> > When collapse it does not show the closing tag
> 
> Do you mean the closing `}` is "inside" the collapsed range instead of outside? I don't think is related to the indenting, I think this is always the case.
> 
> VS Code [doesn't support folding partial lines](https://code.visualstudio.com/api/references/vscode-api#FoldingRange) with columns, only lines. This means we have to decide whether to include the last line or not inside the range. If we include it, then it's not rendered when collapsed (as shown here), and if we don't include it then the collapsed methods would have an additional line for just the closing brace.
> 
> Eg. What I'd like to have is:
> 
> ```dart
> myFunction() { [...] }
> ```
> 
> But actually we only have:
> 
> ```dart
> myFunction() { [...]
> ```
> 
> or:
> 
> ```dart
> myFunction() { [...]
> }
> ```
> 
> I'm not really sure if the bottom one is a great improvement over the current one, since it takes more space in the editor just for the closing brace.

Yes, the closing brace is not showing only for the above syntax. 
The same is done in RN (ECMA). it shows the closing brace. see below

<img width="1215" alt="Screen Shot 2019-11-28 at 15 49 44" src="https://user-images.githubusercontent.com/46526800/69798103-cdc3a700-11f6-11ea-9c6a-6979b61095b3.png">

**Comment by DanTup** [comment]
In my testing, these are all consistent - none of them show the closing brace. Here's a standard function:

<img width="226" alt="Screenshot 2019-11-28 at 12 43 49 pm" src="https://user-images.githubusercontent.com/1078012/69807294-dc04c980-11dc-11ea-8ebe-7d07a1ab8615.png">

I don't think what we have now is perfect, but doing it how I'd like requires a VS Code change. I do think it's better without the closing brace than with if it can't go on the same line though - it seems a waste of screen spaces to take up a whole line just to show a `}` that can be inferred. I think there is no value in showing line 634 in your RN sample.

**Comment by DanTup** [comment]
https://github.com/microsoft/vscode/issues/50840 is the relevant VS Code issue for this. If VS Code gets character-level folding, we should be able to have the `}` appear on the same line as the `...` to avoid wasting lines with just closing brackets.

**Comment by DanTup** [comment]
I'm cleaning up the issue tracker closing some issues that are unlikely to be implemented in the foreseeable future.

Since we can't fix this issue without changes to VS Code and it doesn't look likely they will be made soon, I'm going to close this one. If they do implement something, we can re-open it.
